academic_observatory_workflows/workflows/dqc_workflow.py

```diff
@@ -0,0 +1,456 @@
+        self.sensor_dag_ids = sensor_dag_ids
+
+        assert datasets, "No dataset or tables given for this DQC Workflow! Please revise the config file."
+
+    def make_dag(self) -> DAG:
+        """Create a DAG object for the workflow explicitly defining the tasks and task groups.
+
+        :return: The DAG object for the workflow."""
+
+        with self.dag:
+            # Create a group of sensors for the workflow.
+            task_sensor_group = []
+            if self.sensor_dag_ids:
+                with TaskGroup(group_id="dag_sensors") as tg_sensors:
+                    task_sensors = []
+
+                    for ext_dag_id in self.sensor_dag_ids:
+                        sensor = DagRunSensor(
+                            task_id=f"{ext_dag_id}_sensor",
+                            external_dag_id=ext_dag_id,
+                            mode="reschedule",
+                            duration=timedelta(days=7),
+                            poke_interval=int(timedelta(hours=1).total_seconds()),
+                            timeout=int(timedelta(days=3).total_seconds()),
+                        )
+                        print(sensor)
+                        self.add_operator(sensor)
+                        task_sensors.append(sensor)
+
+                    # Add all of the sesnors to tg_sensors with this line.
+                    (task_sensors)
+
+                    task_sensor_group = tg_sensors
+
+            # Add the standard tasks for the workflow
+            # fmt: off
+            task_check_dependencies = PythonOperator(python_callable=self.check_dependencies, task_id="check_dependencies")
+            task_create_dataset = self.make_python_operator(self.create_dataset, "create_dataset")
+            # fmt: on
+
+            # Add each dataset as a task group and perform the checks on the tables in parallel.
+            task_datasets_group = []
+            for dataset_id in list(self.datasets.keys()):
+                with TaskGroup(group_id=dataset_id) as tg_dataset:
+                    tables: List[Table] = make_tables(self.project_id, dataset_id, self.datasets[dataset_id]["tables"])
+
+                    table_tasks = []
+                    for table in tables:
+                        task = self.make_python_operator(
+                            self.perform_data_quality_check,
+                            task_id=f"{table.table_id}",
+                            op_kwargs={f"task_id": f"{table.table_id}", "table": table},
+                        )
+                        table_tasks.append(task)
+
+                    # Add the list of table_tasks to tg_dataset by this line
+                    (table_tasks)
+
+                    task_datasets_group.append(tg_dataset)
+
+            # Link all tasks and task groups together.
+            (task_sensor_group >> task_check_dependencies >> task_create_dataset >> task_datasets_group)
+
```

Comment:
It would be a bit cleaner to separate the parsing of the "datasets" and "tables" from creating the DAG. You could make a function that converts the datasets / tables passed to the workflow into Table dataclass objects etc.